wordpress-mobile/WordPress-iOS
# UL&S: SiteCredentialsViewController: OnePassword button, error label, and Continue button

Ref. #14009 
Auth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/332

While focusing on iOS 12 testing, I discovered the site URL textfield was using an iOS 13-only delegate method. 

**Known bug** the eye icon is not the correct color on the old UI. I will fix this in a future PR.

This PR: 
1. Addresses the iOS 13-only delegate method and replaces it with an iOS 12 alternative.
2. Adds the OnePassword icon to the username field for iOS 12 users.
3. Fixes the OnePassword icon color bug found in the old UI.
4. Adds the Continue button implementation (form validation, error messages, navigates to next screen)

### Screenshots

**iOS 13 SiteCredentialsViewController**
<a href="https://user-images.githubusercontent.com/1062444/88238281-e29df080-cc46-11ea-85fe-1294d03c0c5b.png"><img src="https://user-images.githubusercontent.com/1062444/88238281-e29df080-cc46-11ea-85fe-1294d03c0c5b.png" width="350" /></a>

####  iPhone 5s, iOS 12
| Before | After |
| --- | --- |
|<a href="https://user-images.githubusercontent.com/1062444/88292839-c68a6580-ccbf-11ea-9af2-03a19370f289.png"><img src="https://user-images.githubusercontent.com/1062444/88292839-c68a6580-ccbf-11ea-9af2-03a19370f289.png" width="350" /></a> | <a href="https://user-images.githubusercontent.com/1062444/88292867-cee2a080-ccbf-11ea-8b16-b795ee39f0ad.png"><img src="https://user-images.githubusercontent.com/1062444/88292867-cee2a080-ccbf-11ea-8b16-b795ee39f0ad.png" width="350" /></a> |
| <a href="https://user-images.githubusercontent.com/1062444/88292894-d99d3580-ccbf-11ea-8cf1-20775c014134.png"><img src="https://user-images.githubusercontent.com/1062444/88292894-d99d3580-ccbf-11ea-8cf1-20775c014134.png" width="350" /></a> | <a href="https://user-images.githubusercontent.com/1062444/88292918-e3269d80-ccbf-11ea-8a40-af041f8b12d0.png"><img src="https://user-images.githubusercontent.com/1062444/88292918-e3269d80-ccbf-11ea-8a40-af041f8b12d0.png" width="350" /></a> |


### To test:
It is recommended to test this PR on an iOS 12 device if you have one available. (I do not).

Dev testing notes: to save yourself some time with testing and building, you can optionally point the podfile to use a local build, because there is some hacking around to test this PR if you don't have an iOS 12 device available. Check out the Auth PR branch, point the podfile to `pod 'WordPressAuthenticator', :path => '../WordPressAuthenticator-iOS'`.

#### Test case 1, option A: build on an iOS 12 device
1. `rake dependencies`
2. Build and run.
3. Log In > Continue with WordPress.com. Observe the OnePassword icon color is updated on the old screen. Go back.
4. Log In > Site Address > enter a valid URL > Continue.
5. On the username / password screen, the OnePassword icon should appear in the username field. Tap it and verify the OnePassword flow works.

#### Test case 1, option B: hack it and test on an iOS 12 simulator
1. Check out the Auth PR wordpress-mobile/WordPressAuthenticator-iOS#332
2. Modify the WPiOS podfile to run Auth locally
3. Comment out the OnePasswordEnabled check 
4. Log In > Continue with Continue with WordPress.com. Observe the OnePassword icon color is updated on the old screen. Go back.
5. Log In > Site Address > enter a valid URL > Continue.
6. Username / password screen > tap the OnePassword icon. It will display an empty share sheet.

#### Test case 2
Run this on an iOS 13 device or sim. Your choice.

1. Reset your environment
2. `rake dependencies`
3. Build and run
4. Log In > Site Address > enter a valid URL > Continue.
5. The username field should not have the OnePassword icon.
6. Enter a fake username / password and see the error message display.
7. Enter a valid username / password and confirm you are navigated to the next screen.

### Do not merge until
1. `.siteAddress` feature flag is disabled
2. Authenticator pod released
3. podfile updated to point to latest Auth release

PR submission checklist:
